mava/utils/checkpointing.py

```diff
@@ -182,25 +175,24 @@ def restore_learner_state(
         # Dictionary of the restored learner state
         restored_learner_state_raw = restored_checkpoint["learner_state"]
 
-        # Restore the learner state type and check it matches the input type
-        restored_learner_state_type = locate(restored_checkpoint["type"])
-        assert restored_learner_state_type == type(unreplicated_input_learner_state)
-
-        # We base the new learner state on the input learner state
-        new_learner_state = copy(unreplicated_input_learner_state)
-
-        if restore_params:
-            new_learner_state = new_learner_state._replace(
-                params=Params(**FrozenDict(restored_learner_state_raw["params"])),
-            )
-
-        if restore_hstates and restored_learner_state_type == RNNLearnerState:
-            new_learner_state = typing.cast(RNNLearnerState, new_learner_state)  # for mypy
-            new_learner_state = new_learner_state._replace(
-                hstates=HiddenStates(**FrozenDict(restored_learner_state_raw["hstates"])),
-            )
+        # Check the type of `input_params` for compatibility.
+        # This is a sanity check to ensure correct handling of parameter types.
+        # In Flax 0.6.11, parameters were typically of the `FrozenDict` type,
+        # but in cu versions, a regular dictionary is used.
```

Comment:
```suggestion
        # but in current versions, a regular dictionary is used.
```
Is this correct?

Reply:
either current version or later versions